rust/src/interpreter.rs

```diff
@@ -831,6 +831,7 @@ where
         self.gas_left.consume(cost)?;
 
         let data = self.memory.get_mut_slice(offset, len, &mut self.gas_left)?;
+        topics.reverse();
```

Comment:
That works with the implementation where Stack holds an array. However with the "simple" implementation using a Vec this is not going to work. I guess this would work with VecDeque. I could try it out at a later point but currently I don't see too much value in it especially because it makes the code harder to reason about because usually data grows from front to back.